lib/client.js

```diff
@@ -347,12 +352,32 @@ export default class Client {
       this.ready = true
       this._metadata = data.metadata
       this._context = data.context
+      // Add an observer.  This will be called when the idleState changes
+      this._removeObserverCallback = this._idleState.addObserver(state => {
+        if (state === 'active') {
+          // ... application is now active, timeout clock is reset to zero
+          // Send a postMessage indicating liveness + possibly a payload too
+          this.postMessage('session.live')
+        } else if (state === 'idle') {
+          // ... application is now idle, timeout clock is ticking
+          // Send a postMessage indicating sessionTimeout + possibly a payload too
+          this.postMessage('session.idle')
+        }
+      })
     }, this)
 
     this.on('context.updated', (context) => {
       this._context = context
     }, this)
 
+    // If the host reaches out to the app via ZAF, it will make something like a 'poll.getIdleState' request.
+    // We should respond with the current state of the iframe.
+    this.on('poll.getIdleState', (data) => {
+      // send a postMessage responding with the current state
+      const currentState = this._idleState.state
+      this.postMessage('poll.getIdleState.response', { idleState: currentState })
+    }, this)
```

Comment:
That's correct @sarmadsangi. 

It would be costly for a host to stay polling each app per app runtime in the different locations around the host product UI to fetch the idle state.

Therefore if an app sends a `postMessage` to the host, in which @token-cjg implemented below:

```js
if (state === 'active') {
  // ... application is now active, timeout clock is reset to zero
  // Send a postMessage indicating liveness + possibly a payload too
  this.postMessage('session.live');
} else if (state === 'idle') {
  // ... application is now idle, timeout clock is ticking
  // Send a postMessage indicating sessionTimeout + possibly a payload too
  this.postMessage('session.idle');
}
```

That would simply pass information to the host only when needed (**I.e:** when an agent is performing some task within an app iframe).

Reply:
Poll was perhaps a bad choice of language.  The way I saw this working is that ZAF does need a way to query what the idleState of an app is.  This could be wrong though, maybe we just want the flow of information to be purely from zaf_sdk to zaf on state change.  Created a new commit here regardless, let me know what you think https://github.com/zendesk/zendesk_app_framework_sdk/pull/145/commits/765950b6bece33115d9373f0dd359ceacb5ae0dd .